maintainers/scripts/activity/activity.bb

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env bb
+
+;; to get a repl
+;; - Terminal
+;;       $ nix-shell -p babashka curl --run "bb"
+;;       user=> (load-file "activity.bb")
+;;       user=> (pprint (file-query :org-repos {:org "nixos"}))
+;; - IDE support
+;;       $ nix-shell -p babashka curl --run "bb nrepl-server"
+;;   - connect to 127.0.0.1:1667 with emacs/cider or vscode/calva
+
+;; # Configuration
+
+(require '[babashka.curl :as curl]
+         '[babashka.cli :as cli])
+
+(def wd (.getParentFile (fs/file *file*)))
+
+(def graphql-path
+  "Make working directory overridable"
+  (str
+   (or (System/getenv "NIXOS_ACTIVITY_GQL_PATH")
+       wd)))
+
+(def maintainers-path
+  "Make maintainers path overridable"
+  (str
+   (or (System/getenv "NIXOS_MAINTAINERS_PATH")
+       (fs/canonicalize
+        (fs/file wd "../..")))))
+
+;; (println "graphql path:" graphql-path)
+
+(defn get-access-token
+  "GitHub access token lookup"
+  []
+  (or (System/getenv "ACCESS_TOKEN")
+      (try
+        (slurp (fs/file (fs/home) ".config/nix/maintainer-activity-access-token"))
+        (catch Exception e nil))
+      (try
+        (->> (line-seq (io/reader (fs/file (fs/home) ".config/nix/nix.conf")))
+             (map #(re-matches #"access-tokens\W*=.*github.com=(\w*)" %))
+             (map second)
+             (filter some?)
+             first)
+        (catch Exception e nil))
+      (do (.println System/err "Please set GitHub Access Token
+  - ACCESS_TOKEN=<token>
+  - ~/.config/nix/maintainer-activity-access-token <token>
+  - ~/.config/nix/nix.conf access-tokens = github.com=<token>
+")
+          (System/exit 1))))
+
+;; ## nixos org and its repositories
+
+(comment
+  ;; to regenerate org id
+  (unwrap (file-query :org-repos {:org "nixos"})
+          [:organization :id]))
+
+(def org-id "O_kgDOAAdwkA")
+
+(def repo-map (edn/read (java.io.PushbackReader.
+                         (io/reader (io/file graphql-path
+                                             "repos.edn")))))
+
+(def interesting-repo?
+  (into #{}
+        (map repo-map)
+        #_["nix" "nixpkgs"] ; to scope down which repos to look at
```

Comment:
Why is this commented out? I don't really think we care if someone is active in other repos.

Reply:
Well, right now we're looking at all repos in NixOS. Are you saying we don't care about most of them and should throw out somebody who's just contributed to other nixos repos?

Reply:
There is no reason to assume that if someone contributes to other projects that are related to nix that they're interested in nixpkgs. If they're interested in nixpkgs, they will contribute to nixpkgs.

Reply:
Or at least, I think we should only care about nixpkgs at the start. Otherwise this script becomes way more complicated than it should.

If there are cases where people contribute in other nix projects and still want to be considered active at nixpkgs, we can then add this logic of looking at other places. This is my opinion.

Reply:
> Otherwise this script becomes way more complicated than it should.

How so? Can you think of a case that would be complicated?

Making this decision feels like a political one, and I don't feel comfortable making it as part of this PR, but I can implement a `--nixpkgs-only` flag, if to make it easier to play around with this option.

Reply:
I've added a `--nixpkgs-only` to `latest-issue-comments`, for now.